src/data/mystery-encounters/encounters/choice-of-balance-encounter.ts

```diff
@@ -0,0 +1,393 @@
+
+const namespace = "mysteryEncounter:choice_of_balance";
+
+function generateRewards(numOptions: number): RewardOption[] {
+  const negativeRewards = getNegativeRewards(numOptions);
+  const positiveRewards = getPositiveRewards(numOptions);
+  const rewardArray = [];
+  for (let i = 0; i < negativeRewards.length; i++) {
+    rewardArray.push(new RewardOption(negativeRewards[i], positiveRewards[i]));
+  }
+
+  return rewardArray;
+}
+
+export const ChoiceOfBalanceEncounter: IMysteryEncounter = MysteryEncounterBuilder
+  .withEncounterType(MysteryEncounterType.CHOICE_OF_BALANCE)
+  .withEncounterTier(MysteryEncounterTier.GREAT)
+  .withIntroSpriteConfigs([
+    {
+      spriteKey: "200",
+      fileRoot: "pokemon",
+      hasShadow: true,
+      x: 4,
+      y: -16,
+      scale: 1,
+      disableAnimation: true // Re-enabled after option select
+    },
+    {
+      spriteKey: "amulet_coin",
+      fileRoot: "items",
+      hasShadow: true,
+      x: 35,
+      y: -3,
+      scale: 0.75,
+      isItem: true,
+    },
+    {
+      spriteKey: "cracked_pot",
+      fileRoot: "items",
+      hasShadow: true,
+      x: 20,
+      y: -14,
+      scale: 0.75,
+      isItem: true,
+    },
+    {
+      spriteKey: "griseous_core",
+      fileRoot: "items",
+      hasShadow: true,
+      x: -6,
+      y: -1,
+      scale: 0.75,
+      isItem: true,
+    }
+  ])
+  .withHideIntroVisuals(false)
+  .withSceneWaveRangeRequirement(10, 180) // waves 10 to 180
+  .withIntroDialogue([
+    {
+      text: `${namespace}_intro_message`,
+    }
+  ])
+  .withOnInit((scene: BattleScene) => {
+    const encounter = scene.currentBattle.mysteryEncounter;
+    encounter.misc = [];
+    console.log(encounter);
+
+    const options = 3;
+
+    const rewardsArray = generateRewards(options);
+
+    for (let i = 0; i < rewardsArray.length; i++) {
+      rewardsArray[i].generateStats();
+      rewardsArray[i].generateAuras();
+      if (rewardsArray[i].negativeStat > 0) {
+        if (rewardsArray[i].negativeOption === NegativeRewards.ENEMY_STATS) {
+          encounter.setDialogueToken("negativeEnemySTAT", getAuraName(rewardsArray[i].negativeStat));
+        } else if (rewardsArray[i].negativeOption === NegativeRewards.PLAYER_STATS) {
+          encounter.setDialogueToken("negativePlayerSTAT", getAuraName(rewardsArray[i].negativeStat));
+        }
+      }
+      if (rewardsArray[i].positiveStat > 0) {
+        if (rewardsArray[i].positiveOption === PositiveRewards.ENEMY_STATS) {
+          encounter.setDialogueToken("positiveEnemySTAT", getAuraName(rewardsArray[i].positiveStat));
+        } else if (rewardsArray[i].positiveOption === PositiveRewards.PLAYER_STATS) {
+          encounter.setDialogueToken("positivePlayerSTAT", getAuraName(rewardsArray[i].positiveStat));
+        }
+      }
+      encounter.misc.push(rewardsArray[i]);
+    }
+
+    encounter.setDialogueToken("dynamic1", rewardsArray[0].generateMessage());
```

Comment:
minor nit: can we have the names of these tokens be more descriptive to their function? "option1Tooltip" or something like that